micrometer-core/src/main/java/io/micrometer/core/instrument/binder/undertow/UndertowMetrics.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright 2017 Pivotal Software, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.core.instrument.binder.undertow;
+
+import io.micrometer.core.instrument.FunctionCounter;
+import io.micrometer.core.instrument.FunctionTimer;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.TimeGauge;
+import io.micrometer.core.instrument.binder.MeterBinder;
+import io.micrometer.core.lang.NonNullApi;
+import io.micrometer.core.lang.NonNullFields;
+import io.undertow.server.handlers.MetricsHandler;
+import io.undertow.servlet.api.MetricsCollector;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.function.ToDoubleFunction;
+import java.util.function.ToLongFunction;
+
+import static io.undertow.server.handlers.MetricsHandler.MetricResult;
+
+/**
+ * {@link MetricsHandler} to collect metrics for {@link io.undertow.Undertow}
```

Comment:
Some information about how to use this class in the JavaDoc would be good. For instance, from the test code, it seems this needs to be set as the metricsCollector on the DeploymentInfo, and it probably needs to be bound after `Undertow` is started?